netboot/main.go

```diff
@@ -165,15 +171,25 @@ func boot(ifname string, dhcp dhcpFunc) error {
 	}
 	debug("DHCP: boot file URL is %s", bootfile)
 	// check for supported schemes
-	if !strings.HasPrefix(bootfile, "http://") {
-		return fmt.Errorf("DHCP: can only handle http scheme")
+	scheme, err := getScheme(bootfile)
+	if scheme == "" || err != nil {
+		return errors.New("DHCP: can only handle http/https scheme")
 	}
 
+	client, err := getClientForBootfile(bootfile)
+	if err != nil {
+		return fmt.Errorf("DHCP: cannot get http client for %s: %v", bootfile, err)
```

Comment:
we may extend to non-HTTP in the future

```suggestion
		return fmt.Errorf("DHCP: cannot get client for %s: %v", bootfile, err)
```